### PR TITLE
fix: ember deprecation warnings

### PR DESCRIPTION
**Pull request by MitanOmar**

### CODE REVIEW

File: app/components/async-list/template.hbs [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <div class="empty">
     <LoadingIcon />
   </div>
-{{else if @data.last.isError}}
+{{else if (not @data)}}
```
Comment (derrabauke) [inline app/components/async-list/template.hbs:5]: `@data.isError` is a official [task state](https://ember-concurrency.com/api/TaskInstance.html) which is also available in here. It's also mentioned [here](https://ember-resources.pages.dev/funcs/util_ember_concurrency.trackedTask):

> (and all other properties on a [TaskInstance](https://ember-concurrency.com/api/TaskInstance.html))

File: tests/integration/components/async-list/component-test.js [hunk 2]
```diff
@@ -47,7 +47,7 @@ module("Integration | Component | async list", function (hooks) {
   });
 
   test("shows error message", async function (assert) {
-    this.set("data", { last: { isError: true } });
+    this.set("data", false);
```
Comment (derrabauke) [inline tests/integration/components/async-list/component-test.js:50]: We have to adapt this to `isError` on the first level of the `data` object.

File: app/components/customer-visible-icon/component.js [hunk 3]
```diff
@@ -1,9 +1,5 @@
-import Component from "@ember/component";
-import { attributeBindings } from "@ember-decorators/component";
-import classic from "ember-classic-decorator";
+import Component from "@glimmer/component";
 
-@classic
-@attributeBindings("title")
 export default class CustomerVisibleIcon extends Component {
   title = "This project's comments are visible to the customer";
```
Comment (derrabauke) [inline app/components/customer-visible-icon/component.js:4]: Can't we make this a template only component by moving the title directly into the template?

File: app/services/tracking.js [hunk 4]
```diff
@@ -135,10 +134,10 @@ export default class TrackingService extends Service {
...
       let task = "Unknown Task";
 
-      if (get(this, "activity.task.content")) {
-        const c = get(this, "activity.task.project.customer.name");
-        const p = get(this, "activity.task.project.name");
-        const t = get(this, "activity.task.name");
+      if (this.activity.task.content) {
+        const c = this.activity.task.get("project.customer.name");
+        const p = this.activity.get("task.project.name");
+        const t = this.activity.get("task.name");
```
Comment (derrabauke) [inline app/services/tracking.js:140]: This seems a little inconsistent. Why don't we need the get for the first `task` access?

Comment (MitanOmar) [reply app/services/tracking.js:140]: it was throwing errors without it

File: app/users/index/controller.js [hunk 5]
```diff
@@ -1,106 +1,101 @@
...
-
-  reset() {
-    /* istanbul ignore next */
-    this.resetQueryParams();
-  },
+  }
+
+  _fetchData = trackedTask(this, this.data, () => {
+    return [this.supervisor, this.search, this.ordering, this.active];
+  });
```
Comment (derrabauke) [inline app/users/index/controller.js:27]:
```suggestion
  _fetchData = trackedTask(this, this.data, () => [this.supervisor, this.search, this.ordering, this.active]);
```

File: app/users/index/controller.js [hunk 6]
```diff
@@ -1,106 +1,101 @@
...
-    }
-  ),
-
-  setup() {
+export default class UsersIndexController extends Controller {
+  queryParams = ["search", "supervisor", "active", "ordering"];
+  @service session;
+  @service router;
+  @service store;
+  @tracked currentUser = this.session.data.user;
```
Comment (derrabauke) [inline app/users/index/controller.js:14]: This is not the same as `currentUser: reads("session.data.user")`

```suggestion
  get currentUser  () {
    return this.session.data.user
  }
```

File: app/users/index/controller.js [hunk 7]
```diff
@@ -1,106 +1,101 @@
...
-    defaultValue: "1",
-    replace: true,
-    refresh: true,
-  },
-  ordering: {
-    defaultValue: "username",
-    replace: true,
-    refresh: true,
-  },
-});
```
Comment (derrabauke) [inline app/users/index/controller.js:30]: The previous `ember-parachute` implementation had the `refresh` and `replace` parameters set to true. Did you have a look what they do? 

I guess we have to add the [`refreshModel` parameter](https://guides.emberjs.com/release/routing/query-params/#toc_opting-into-a-full-transition) on the refactored route.

File: app/users/index/controller.js [hunk 8]
```diff
@@ -1,106 +1,101 @@
...
-      ...(this.get("currentUser.isSuperuser")
+      ...(this.currentUser.isSuperuser
         ? {}
         : {
-            supervisor: this.get("currentUser.id"),
+            supervisor: this.currentUser.id,
           }),
     });
-  }).restartable(),
+  }
```
Comment (derrabauke) [inline app/users/index/controller.js:88]: We could implement this with the route (model) and the native query params, too. Or use 
 [ember-async-data](https://github.com/tracked-tools/ember-async-data). This whole controller feels so overcomplicated. :man_shrugging:

**Review by derrabauke — APPROVED** [review]
LGTM :eyeglasses:

**Pull request merged**